tools/wake-format/emitter.cpp

```diff
@@ -71,6 +90,16 @@ auto Emitter::rhs_fmt() {
   // clang-format on
 }
 
+auto position_comment() {
+  // ws COMMENT -> ws COMMENT nl
+  // ws nl COMMENT -> nl COMMENT nl
+  return fmt().fmt_if(has_comment_next,
```

Comment:
You currently duplicate the `has_comment_next` here and it use site, pick one and stick with it. I think here is better than there.

Reply:
This is needed at the callsite since we should only do `join(fmt().token(TOKEN_COMMENT))` if `position_comment()` evaluates.

I also agree that `position_comment` should be the place where the check lives.

What do you think about having position_comment take `fmt().token(TOKEN_COMMENT)` as a parameter?

Then the callsite becomes

```
fmt()
  ...
  .position_comment(fmt().token(TOKEN_COMMENT))
  ...
```

it also gives a slightly stronger enforcement of the fact that the caller needs to consume the comment

Reply:
Yeah that's better...that said if this is always how this works position_comment should just consume the token

Reply:
yeah. there was a reason that I split it out, but I'm not seeing it anymore. Maybe that was part of a previous issue.
I'll lift it back into the function and we can always pull it back out later if needed.